src/PowerShellEditorServices/Analysis/AnalysisService.cs

```diff
@@ -467,22 +511,51 @@ private PSObject[] InvokePowerShell(string command, IDictionary<string, object>
             return await task;
         }
 
-        private bool VerifyPSScriptAnalyzerAvailable()
+        /// <summary>
+        /// Create a new runspace pool around a PSScriptAnalyzer module for asynchronous script analysis tasks.
+        /// This uses the PowerShell module API to load the first PSScriptAnalyzer module on the module path with
+        /// with the required minimum version and then creates a runspace pool with that module loaded in the initial
+        /// session state.
+        /// </summary>
+        /// <returns>A runspace pool with PSScriptAnalyzer loaded for running script analysis tasks.</returns>
+        private static RunspacePool CreatePssaRunspacePool(out PSModuleInfo pssaModuleInfo)
         {
             using (var ps = System.Management.Automation.PowerShell.Create())
             {
+                // Encode an equivalent of the PowerShell pipeline:
+                //   Get-Module -ListAvailable -Name "PSScriptAnalyzer" | Where-Object -Property "Version" -ge "1.16"
```

Comment:
I feel like we should address that in an issue and have a proper discussion - just so we know what scenarios there are for version-locking vs flexible drop-in.

Reply:
If I remember correctly, the extension used to install PSSA in its AppData folder but does not do that any more. Is that correct?
I don't know much about the compatibility of PSSA before 1.16.1 but the version of it should not matter too much. For example due to one of the recent PRs, PSES now has a new rule from 1.17.0 but it is no problem if one has only 1.16.1 because PSSA just ignores rules in the pssa settings that it  does not recognize.

Reply:
Currently we install PSSA to a directory internal to the extension itself, and when the extension is loaded it adds that directory to the end of the module path in the internally hosted PowerShell session.

Reply:
The question is really whether we always take the latest PSSA (i.e. PSSA 17.1) or we use a minimum version.

Given the recent re-release of PSSA to 17.1, I would imagine it benefits us to only use a minimum version, so that we don't block an update of PSSA unnecessarily (by block I mean, users would only get that new release when we do our release).

Reply:
So by that logic, I suppose I'm suggesting we should:
* Use a minimum version
* Set the minimum version to whatever the currently released version of PSSA is at our release time

Ideally we could tool the second point so that we don't have to update the version in source here, but I could see that taking some doing.

Reply:
When I meant latest - I meant like... sort the results we get from `Get-Module` and then grab the newest on the machine like it use to be.

Reply:
Because as it is now, we don't really know what `.FirstOrDefault` a few lines down is going to give us. Could be the latest on the machine, could be the oldest on the machine, or a random one.

Reply:
Does `Get-Module -ListAvailable` output in the order that `Import-Module` would import from?

If it does, then I think there's value in it.  That conforms well with the original intention of the change, to give the user more control over what version is used. 

If it doesn't, then we should probably stick to the highest version installed until we can drop v3/4 and use the better `initialsessionstate.ImportModule` API.

Reply:
`Get-Module -ListAvailable` looks like it goes over the module path in order (i.e. respects the module path).

Actual method that gets all the modules:
https://github.com/PowerShell/PowerShell/blob/4192463ce6119ec298459ea60cd12f5c2fa1f07c/src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs#L1001-L1045

ModuleIntrinsics method it uses to enumerate the module path:
https://github.com/PowerShell/PowerShell/blob/4192463ce6119ec298459ea60cd12f5c2fa1f07c/src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs#L1009-L1031

Also worth noting that the newer module import API I wanted to move to before we noticed the incompatibility [also uses `Get-Module -ListAvailable` internally](https://github.com/PowerShell/PowerShell/blob/ebc7cdf847ef39c8d71fd0f69849bf59f498111b/src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs#L3969-L3984).

Reply:
I'm assuming of course that older PowerShell versions also have this behaviour...

Reply:
I've looked at Windows PowerShell 5.1 and I don't think it guarantees a module path ordering for `Get-Module -ListAvailable`. In fact, neither does `InitialSessionState.ImportPSModule(ModuleSpecification[])` or `InitialSessionState.ImportPSModule(string[])`. I don't there is any API for module imports that respects the module path. We would need to search the path ourselves...

Reply:
Welp that complicates things.

The benefit of obeying `Import-Module` ordering logic is predictability.  If we lose that it feels more like a gamble which module you're going to import.  Do we know of anyone who was specifically trying to downgrade PSSA?  It isn't super common for folks to build against PSSA as far as I know, so I don't think issues with it will be as common as say Newtonsoft.  If we haven't had any specific complaints we may want to switch back to importing the latest only.

Reply:
Why not just select the version number and pipe to sort to pick the latest one?

Reply:
The original idea was to use at least the version we shipped with the extension but allow users to use a "newer" version if installed on their machine.  So, using latest sounds good to me.

Reply:
Yeah I think the use-latest approach is the best way -- at least `InitialSessionState.ImportPSModulesFromPath` seems to have a sane behaviour.

Reply:
I've opened an issue for our above discussion BTW:
https://github.com/PowerShell/PowerShell/issues/7020